spanner/spanner_samples.rb

```diff
@@ -968,6 +968,267 @@ def update_using_batch_dml project_id:, instance_id:, database_id:
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  example_bool = true
+  sql_query = "SELECT VenueId, VenueName, OutdoorVenue FROM Venues
+               WHERE OutdoorVenue = @outdoor_venue"
+
+  params      = { outdoor_venue: example_bool }
+  param_types = { outdoor_venue: :BOOL }
+
+  client.execute(sql_query, params: params, types: param_types).rows.each do |row|
+    puts "#{row[:VenueId]} #{row[:VenueName]} #{row[:OutdoorVenue]}"
+  end
+  # [END spanner_query_with_bool_parameter]
+end
+
+def query_with_bytes project_id:, instance_id:, database_id:
+  # [START spanner_query_with_bytes_parameter]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  example_bytes = StringIO.new "Hello World 1"
+  sql_query = "SELECT VenueId, VenueName FROM Venues
+               WHERE VenueInfo = @venue_info"
+
+  params      = { venue_info: example_bytes }
+  param_types = { venue_info: :BYTES }
+
+  client.execute(sql_query, params: params, types: param_types).rows.each do |row|
+    puts "#{row[:VenueId]} #{row[:VenueName]}"
+  end
+  # [END spanner_query_with_bytes_parameter]
+end
+
+def query_with_date project_id:, instance_id:, database_id:
+  # [START spanner_query_with_date_parameter]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  example_date = "2019-01-01"
+  sql_query = "SELECT VenueId, VenueName, LastContactDate FROM Venues
+               WHERE LastContactDate < @last_contact_date"
+
+  params      = { last_contact_date: example_date }
+  param_types = { last_contact_date: :DATE }
+
+  client.execute(sql_query, params: params, types: param_types).rows.each do |row|
+    puts "#{row[:VenueId]} #{row[:VenueName]} #{row[:LastContactDate]}"
+  end
+  # [END spanner_query_with_date_parameter]
+end
+
+def query_with_float project_id:, instance_id:, database_id:
+  # [START spanner_query_with_float_parameter]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  example_float = 0.8
+  sql_query = "SELECT VenueId, VenueName, PopularityScore FROM Venues
+               WHERE PopularityScore > @popularity_score"
+
+  params      = { popularity_score: example_float }
```

Comment:
Why not put the value of `example_float` in place?

Reply:
The goal of this addition to the sample is to show how to create a variable set which is set to the value of given datatype and then use it as a parameter in a Cloud Spanner query. In Ruby this very straightforward :), so I can update the PR if you prefer the parameter values in place.

Reply:
Apologies for the delay. I would recommend doing in place parameter values instead.

Please update.